crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -367,6 +378,168 @@ impl mandate::MandateBehaviour for types::PaymentsAuthorizeData {
     }
 }
 
+pub async fn update_connector_customer_in_customers(
+    state: &AppState,
+    connector: &api::ConnectorData,
+    customer: &Option<storage::Customer>,
+    connector_customer_map: Option<serde_json::Map<String, serde_json::Value>>,
+    connector_cust_id: Option<String>,
+) -> RouterResult<Option<String>> {
+    let mut connector_customer = match connector_customer_map {
+        Some(cc) => cc,
+        None => serde_json::Map::new(),
+    };
+    connector_cust_id.clone().map(|cc| {
+        connector_customer.insert(
+            connector.connector_name.to_string(),
+            serde_json::Value::String(cc),
+        )
+    });
+    let db = &*state.store;
+    let update_customer = storage::CustomerUpdate::ConnectorCustomer {
+        connector_customer: Some(serde_json::Value::Object(connector_customer)),
+    };
+    match customer {
+        Some(customer) => {
+            db.update_customer_by_customer_id_merchant_id(
+                customer.customer_id.to_owned(),
+                customer.merchant_id.to_owned(),
+                update_customer,
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to update CustomerConnector in customer")?;
+            Ok(connector_cust_id)
+        }
+        None => Ok(connector_cust_id),
+    }
+}
+
+type CreateCustomerCheck = (
+    bool,
+    Option<String>,
+    Option<serde_json::Map<String, serde_json::Value>>,
+);
+pub fn should_call_connector_create_customer(
+    state: &AppState,
+    connector: &api::ConnectorData,
+    customer: &Option<storage::Customer>,
+) -> RouterResult<CreateCustomerCheck> {
+    let connector_name = connector.connector_name.to_string();
+    let connector_customer_filter = state
+        .conf
+        .connector_customer
+        .connector_list
+        .contains(&connector.connector_name);
+    if connector_customer_filter {
```

Comment:
Avoid deep nesting function, split into small function and write a comments too.

Reply:
i will add comment , its just a customer and call_customer check , have divided already in small fn

Reply:
Add comments for None and else cases.